acdcli/api/content.py

```diff
@@ -61,57 +62,63 @@ class ContentMixin(object):
     """Implements content portion of the ACD API."""
 
     def create_folder(self, name: str, parent=None) -> dict:
-        body = {'kind': 'FOLDER', 'name': name}
-        if parent:
-            body['parents'] = [parent]
-        body_str = json.dumps(body)
+        while True:
+            body = {'kind': 'FOLDER', 'name': name}
+            if parent:
+                body['parents'] = [parent]
+            body_str = json.dumps(body)
 
-        acc_codes = [http.CREATED]
+            acc_codes = [http.CREATED]
 
-        r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            if r.status_code in RETRY_CODES: continue  # the fault lies not in our stars, but in amazon
```

Comment:
This can lead theoretically to an endless loop. A max retry count could avoid that.
In conditions were the amazon server gives always a 500 back this will never end and will useless only stress the client and the server.